backend/tracim_backend/lib/webdav/resources.py

```diff
@@ -68,143 +70,224 @@ def wrapper(self: "_DAVResource", *arg, **kwarg) -> typing.Callable:
                 # We decide to show only first same name workspace (in order of get_all() which is workspace
                 # id order). So, same name workspace are not supported from webdav but doesn't cause
                 # much trouble: only one workspace is accessible without any issue.
                 continue
-            members_names.append(webdav_convert_file_name_to_display(workspace.label))
-        return members_names
-
-    @webdav_check_right(is_user)
-    def getMember(self, label: str) -> typing.Optional[DAVCollection]:
-        """
-        This method returns the child Workspace that corresponds to a given name
+            else:
+                members_names.append(workspace_label)
+                members.append(workspace)
+        return members
 
-        Though for perfomance issue, we're not using this function anymore
-        """
-        try:
-            workspace = self.workspace_api.get_one_by_label(label)
-            # fix path
-            workspace_path = "%s%s%s" % (
-                self.path,
-                "" if self.path == "/" else "/",
-                webdav_convert_file_name_to_display(workspace.label),
-            )
-            # return item
-            return WorkspaceResource(
-                path=workspace_path,
-                environ=self.environ,
-                workspace=workspace,
-                tracim_context=self.tracim_context,
-                label=workspace.label,
-            )
-        except AttributeError:
-            return None
+    def _generate_child_workspace_resource(
+        self, parent_path: str, child_workspace: Workspace
+    ) -> "WorkspaceResource":
+        workspace_label = webdav_convert_file_name_to_display(child_workspace.filemanager_filename)
+        path = add_trailing_slash(parent_path)
+        workspace_path = "{}{}".format(path, workspace_label)
+        return get_workspace_resource(
+            path=workspace_path,
+            environ=self.environ,
+            workspace=child_workspace,
+            tracim_context=self.tracim_context,
+            label=workspace_label,
+        )
 
-    @webdav_check_right(is_trusted_user)
+    # Container methods
     def createEmptyResource(self, name: str):
-        """
-        This method is called whenever the user wants to create a DAVNonCollection resource (files in our case).
+        raise NotImplementedError()
 
-        There we don't allow to create files at the root;
+    def createCollection(self, label: str) -> "FolderResource":
+        raise NotImplementedError()
+
+    def getMemberNames(self) -> List[str]:
         """
-        raise DAVError(HTTP_FORBIDDEN, contextinfo="Not allowed to create new root")
+        This method returns the names (here workspace's labels) of all its children
 
-    @webdav_check_right(is_trusted_user)
-    def createCollection(self, name: str):
+        Though for perfomance issue, we're not using this function anymore
```

Comment:
```suggestion
        This method returns the names (here, the labels of workspaces) of all its children.
        For performance concerns, we're not using this function anymore
```

Can we remove it then? And from the interface defined earlier too?

Reply:
In fact, i don't know where this comment came from, and i'm not completely sure this method is not used by wsgidav (it's, so i keep implementation of both getMember and getMemberList for all resources. I suggest to make a comment about this unclear point.

Reply:
i add a more clear info comment about this point.